community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/StatisticsBackedCardinalityModel.scala

```diff
@@ -42,6 +43,10 @@ class StatisticsBackedCardinalityModel(queryGraphCardinalityModel: QueryGraphCar
     case RegularQueryProjection(_, QueryShuffle(_, None, Some(limit: IntegerLiteral))) =>
       Cardinality.min(in, limit.value.toDouble)
 
+    // Normal projection with LIMIT
+    case RegularQueryProjection(_, QueryShuffle(_, None, Some(unknownLimit))) =>
+      Cardinality.min(in, GraphStatistics.DEFAULT_LIMIT_CARDINALITY)
+
```

Comment:
I believe it's a balancing act, and not clear cut which is the best option. Given that `SKIP` and `LIMIT` are common ways of paging through results, the compiler is likely to see the same query multiple times where the only difference is the literals in the LIMIT/SKIP values.
Not auto-parameterising these would result in many more re-compilations of paging queries.

The user can very simply fix the problem by not hand-coding string and properly using parameters (which turns off auto-parameterisation), and then we do use the literal values.

WDYT? Should we auto-parameterise these values?